wordpress-mobile/WordPress-iOS
# Fix a Core Data concurrency issue when CrashLogging is called from a background thread

Fixes #20836

## Description

This PR resolves an issue causing the app to crash when `CrashLogging.logError` is called from a background thread. More details in #20836.

## Test Instructions

I couldn't find a way to reproduce this crash in production, but it's easy to trigger this issue programmatically.

1. First, enable `-com.apple.CoreData.ConcurrencyDebug 1` launch argument.
2. Paste the following code at the end of AppDelegate `didFinishLaunchingWithOptions` method.
```swift
UserSettings.userHasForcedCrashLoggingEnabled = true
let crashLogging = Self.crashLogging
DispatchQueue.global().async {
   crashLogging?.logMessage("Hello World")
}
```
3. Build and run the Jetpack app and ensure you are logged in.
4. Expect the app to crash.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:
